### PR TITLE
Import configs using raw JSON payload

### PR DESCRIPTION
**Pull request by gulopine**
As a bonus, using a raw JSON body instead of a multipart file upload, we can use the existing API client, rather than relying on restler for just this one case. So I've removed that from the project as well.

### CODE REVIEW

File: lib/commands/connect/shared.js [hunk 1]
```diff
@@ -55,7 +55,9 @@ var request = exports.request = function(token, method, path, data){
   // add data to body for POST/PUT requests
   if (data !== undefined && ['POST', 'PUT', 'PATCH'].indexOf(method) !== -1) {
     withBody = true;
-    data = JSON.stringify(data);
+    if(typeof data !== 'string') {
```
Comment (gulopine) [inline lib/commands/connect/shared.js]: Only stringify data if it's not already a string.

File: lib/commands/connect/import.js [hunk 2]
```diff
@@ -23,18 +22,10 @@ module.exports = {
...
-                headers: {
-                  Authorization: 'Bearer ' + context.auth.password
-                },
-                multipart: true,
-                data: {
-                  'config': rest.file(context.args.file, null, stats.size, null, context.args.file)
-                }
-              }).on('complete', function(data) {
+            fs.readFile(context.args.file, 'utf8', function(err, data) {
+              api.request(context.auth.password, 'POST', url, data).then(function(response) {
```
Comment (lambacck) [inline lib/commands/connect/import.js]: I wonder if we should JSON.parse data and not have the jsonify conditional. Provides the benefit of having invalid json detected in the client and keeps a single path through api.request.

Comment (gulopine) [inline lib/commands/connect/import.js]: I wondered about that after I submitted the PR. I definitely don't like testing the type in api.request, so this seems like a reasonable approach.

**Comment by gulopine** [comment]
:eyes:

**Comment by lambacck** [comment]
:+1:

**Pull request merged**